src/components/MCQ/MCQTagOptions.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <div class="filter">
+    <div
+      v-for="[index, valueKeys] in Object.entries(filterSet)"
+      :key="index"
+      class="category"
+    >
+      <h2>{{ index }}</h2>
+      <ul>
+        <li
+          v-for="[key, val] in Object.entries(valueKeys)"
+          :key="key"
+          class="filter-options"
+        >
+          <input type="checkbox" /><label>{{ val }} </label>
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { tags } from "@/types/MCQ";
+import { getAllQuestions } from "../DataAccessLayer";
+function getUniquePropertyValues(
+  array: tags[],
+  property: "course" | "subject" | "system",
+) {
+  return [...new Set(array.map((item) => item[property]))];
+}
+
+const tagSet = getAllQuestions().flatMap((question) => question.tags);
+
+const filterSet = {
+  course: getUniquePropertyValues(tagSet, "course"),
+  subject: getUniquePropertyValues(tagSet, "subject"),
+  system: getUniquePropertyValues(tagSet, "system"),
+};
```

Comment:
This requires multiple iterations of creating a new set with map. Would it be better to just iterate through the array once and add it to the set?
```ts
function getUniquePropertyValues(array: tags[]) {
  const uniqueValues = {
    course: new Set(),
    subject: new Set(),
    system: new Set(),
  };

  array.forEach((item) => {
    uniqueValues.course.add(item.course);
    uniqueValues.subject.add(item.subject);
    uniqueValues.system.add(item.system);
  });

  return {
    course: [...uniqueValues.course],
    subject: [...uniqueValues.subject],
    system: [...uniqueValues.system],
  };
}

const tagSet = getAllQuestions().flatMap((question) => question.tags);

const filterSet = getUniquePropertyValues(tagSet);
```

Reply:
For readability purposes though, your current code seems a bit better